src/create.tsx

```diff
@@ -409,7 +422,12 @@ export const create = <PeerMetadata, TrackMetadata>(
       ) => {
         const broadcastOnDeviceChange = configRef.current.microphone.onDeviceChange ?? "replace";
 
-        if (client.status === "joined" && event.mediaDeviceType === "userMedia" && !pending) {
+        if (
+          client.status === "joined" &&
+          event.mediaDeviceType === "userMedia" &&
+          !pending &&
+          !client.isReconnecting()
+        ) {
```

Comment:
This if is long and it's repeated and I don't know what exatly it means in general, maybe refactor it into a function or something?

Reply:
I've extracted this logic to separate functions `isBroadcastedTrackChanged` and `isBroadcastedTrackStopped`, refactored other fragments in the useSetupMedia hook, and extracted this hook to a separate file. I think the code could be simplified, but I don't want to do it right now.